pkg-manager/plugin-commands-installation/src/link.ts

```diff
@@ -124,7 +124,7 @@ export async function handler (
   const storeControllerCache = new Map<string, Promise<{ dir: string, ctrl: StoreController }>>()
   let workspacePackagesArr
   let workspacePackages!: WorkspacePackages
-  if (opts.workspaceDir) {
+  if (opts.workspaceDir && opts.workspacePackagePatterns) {
     workspacePackagesArr = await findWorkspacePackages(opts.workspaceDir, {
       ...opts,
       patterns: opts.workspacePackagePatterns,
```

Comment:
Maybe just add a default value for patterns in `findWorkspacePackages`. Otherwise the tests have to be updated with `workspacePackagePatterns`, which isn't even used in most cases because `allProjects` is passed.

Reply:
I noticed that above too. https://github.com/pnpm/pnpm/pull/8214#discussion_r1641633616

A lot of tests have a common options setup that defaults values from `@pnpm/config`. Perhaps we could those to `['.', '**']` for tests.